app/frontend/src/utils/types.ts

```diff
@@ -29,3 +29,6 @@ export type ProtoToJsTypes<T> = T extends StringValue.AsObject
 export type SetMutationError = React.Dispatch<
   React.SetStateAction<string | null>
 >;
+
+export type Primitives = string | number | boolean | null;
+export type ServicePrimitives = number | { value: Primitives };
```

Comment:
It's not clear to me what this type is for from the name. 

If it's some kind of protobuf message type, I would expect it to be `Primitives | { value: Primitives }`

But depending on where it is used, maybe better to just use that inline?

Reply:
I don't know the internals enough yet to decide this. I was just looking that at this moment the values being assigned to the variable are either `number` or `{value: Primitives}` . But I can move it back inline.

Reply:
Yeah maybe do inline, that way we don't need to think of a descriptive name. And I don't think it will be needed that much (otherwise we can add it later)

Reply:
I've moved `ServicePrimitives` to an inline type. Or did you mean to also move the `Primitives` type to _useUpdateHostingPreferences.test.tsx_?

Reply:
nope this is good